Backend/Gmfctn/Controllers/AchievementController.cs

```diff
@@ -1,75 +1,90 @@
 ﻿using AutoMapper;
 using Data_;
 using Data_.Dtos;
-using Data_.Interfaces;
-using Data_.Validators;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.EntityFrameworkCore;
 using System;
 using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
+using Services;
+using Services.Interfaces;
+using Microsoft.AspNetCore.Authorization;
 
 namespace Gmfctn.Controllers
 {
     [Route("api/achievement")]
     [ApiController]
+    [Authorize]
     public class AchievementController : ControllerBase
     {
-        private IUnitOfWork UnitOfWork;
-        private readonly IMapper _Mapper;
-        public AchievementController(GmfctnContext Context, IMapper Mapper) {
-            UnitOfWork = new UnitOfWork(Context);
-            _Mapper = Mapper;
+        private readonly IMapper Mapper;
```

Comment:
If you don't use Mapper here then we don't need to declare it, so please remove it

Reply:
Fixed